From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Akhil R <akhilrajeev@nvidia.com>
Date: Wed, 3 Apr 2024 15:03:09 +0530
Subject: [PATCH 002/240] NVIDIA: SAUCE: crypto: Skip lrw(aes) in tcrypt

Skip lrw(aes) from AES test in tcrypt. It is not supported by Tegra SE.
Also enable the success print so that the success message is clear.

Signed-off-by: Akhil R <akhilrajeev@nvidia.com>
Change-Id: I9d5219fe9f77a3bca81dd4fcf38928f14172b6d6
Reviewed-on: https://git-master.nvidia.com/r/c/linux-stable/+/3108523
GVS: Gerrit_Virtual_Submit <buildbot_gerritrpt@nvidia.com>
Reviewed-by: Laxman Dewangan <ldewangan@nvidia.com>

diff --git a/crypto/tcrypt.c b/crypto/tcrypt.c
index e86d1470227c..41916804efef 100644
--- a/crypto/tcrypt.c
+++ b/crypto/tcrypt.c
@@ -1520,10 +1520,10 @@ static int do_test(const char *alg, u32 type, u32 mask, int m, u32 num_mb)
 	case 10:
 		ret = min(ret, tcrypt_test("ecb(aes)"));
 		ret = min(ret, tcrypt_test("cbc(aes)"));
-		ret = min(ret, tcrypt_test("lrw(aes)"));
 		ret = min(ret, tcrypt_test("xts(aes)"));
 		ret = min(ret, tcrypt_test("ctr(aes)"));
 #if !defined(CONFIG_CRYPTO_SUPPORT_TEGRA_TESTS)
+		ret = min(ret, tcrypt_test("lrw(aes)"));
 		ret = min(ret, tcrypt_test("rfc3686(ctr(aes))"));
 		ret = min(ret, tcrypt_test("xctr(aes)"));
 #endif
@@ -2828,7 +2828,7 @@ static int __init tcrypt_mod_init(void)
 		pr_err("one or more tests failed!\n");
 		goto err_free_tv;
 	} else {
-		pr_debug("all tests passed\n");
+		pr_info("all tests passed\n");
 	}
 
 	/* We intentionaly return -EAGAIN to prevent keeping the module,
-- 
2.55.0

